haskell/experimental/private/module.bzl

```diff
@@ -376,14 +376,14 @@ def _collect_module_inputs(module_input_map, directs, dep):
     return all_inputs
 
 def _collect_narrowed_deps_module_files(ctx_label, per_module_transitive_files, dep):
-    direct_cross_library_deps = dep[HaskellModuleInfo].direct_cross_library_deps
+    transitive_cross_library_dep_labels = dep[HaskellModuleInfo].transitive_cross_library_dep_labels.to_list()
```

Comment:
This flattening could be problematic because the depset could be large. I'm still thinking how to avoid it.

Reply:
I reverted this and created a better solution in af97995.